gong/connector.go

```diff
@@ -0,0 +1,81 @@
+package gong
+
+import (
+	"net/http"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/interpreter"
+	"github.com/amp-labs/connectors/common/paramsbuilder"
+
+	"errors"
+
+	"github.com/amp-labs/connectors/providers"
+)
+
+var DefaultModule = paramsbuilder.APIModule{ // nolint: gochecknoglobals
+	Label:   "api/data",
+	Version: "v2",
+}
+
+type Connector struct {
+	BaseURL   string
+	Client    *common.JSONHTTPClient
+	APIModule paramsbuilder.APIModule
+}
+
+func WithCatalogSubstitutions(substitutions map[string]string) Option {
+	return func(params *gongParams) {
+		params.substitutions = substitutions
+	}
+}
+
+func NewConnector(opts ...Option) (conn *Connector, outErr error) {
+	defer common.PanicRecovery(func(cause error) {
+		outErr = cause
+		conn = nil
+	})
+
+	params := &gongParams{}
+	for _, opt := range opts {
+		opt(params)
+	}
+
+	var err error
+
+	params, err = params.prepare()
+	if err != nil {
+		return nil, err
+	}
+
+	// Read provider info
+	providerInfo, err := providers.ReadInfo(providers.Gong, &map[string]string{
+		"workspace": params.Workspace.Name,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	conn = &Connector{
+		Client:    params.client,
+		BaseURL:   providerInfo.BaseURL,
+		APIModule: params.APIModule,
+	}
+
+	conn.setBaseURL(providerInfo.BaseURL)
+	conn.Client.HTTPClient.ErrorHandler = interpreter.ErrorHandler{
```

Comment:
We can't expect our customers to be modifying this code to be adding error handling logic. Can you revise this to:

```
	conn.Client.HTTPClient.ErrorHandler = conn.interpretError
```

Reply:
fixed